iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/CommandDetector.java

```diff
@@ -260,6 +260,10 @@ public CommandDetector build() {
 
   public static class Command implements HasTextRange {
 
+    public List<ArgumentResolution> getResolvedArguments() {
```

Comment:
1. Please move this method after the constructor or `textRange()`.
2. We are breaking encapsulation here, what makes me sad :-( But for now I don't see a better way how to implement :-(

Reply:
Probably, could be removed after improvements to `CommandDetector` that would allow to match command more complexly and without these additional checks. I'll link https://sonarsource.atlassian.net/browse/SONARIAC-1088 in a TODO comment